bi_view_editor/models/ir_model.py

```diff
@@ -0,0 +1,223 @@
+
+            Return all possible join nodes to add new_field to the query
+            containing model_ids.
+        """
+        model_ids = dict([(field['table_alias'],
+                           field['model_id']) for field in field_data])
+        keys = [(field['table_alias'], field['id'])
+                for field in field_data if field.get('join_node', -1) != -1]
+        join_nodes = ([{'table_alias': alias}
+                       for alias, model_id in model_ids.items()
+                       if model_id == new_field['model_id']] + [
+            d for d in self.get_related_fields(model_ids)
+            if (d['relation'] == new_field['model'] and
+                d['join_node'] == -1) or
+            (d['model_id'] == new_field['model_id'] and
+             d['table_alias'] == -1)])
+        return filter(
+            lambda x: 'id' not in x or
+                      (x['table_alias'], x['id']) not in keys, join_nodes)
+
+    @api.model
+    def get_fields(self, model_id):
+        bi_field_domain = [
+            ('model_id', '=', model_id),
+            ("name", "not in", NO_BI_FIELDS),
+            ('ttype', 'not in', [
+                'many2many', "one2many", "html", "binary", "reference"])
+        ]
+        filter_bi_fields = self._filter_bi_fields
+        fields_obj = self.env['ir.model.fields']
+        fields = filter(filter_bi_fields,
+                        fields_obj.sudo().search(bi_field_domain))
+        return sorted([{"id": field.id,
+                        "model_id": model_id,
+                        "name": field.name,
+                        "description": field.field_description,
+                        "type": field.ttype,
+                        "custom": False,
+                        "model": field.model_id.model,
+                        "model_name": field.model_id.name}
+                       for field in fields], key=lambda x: x['description'],
+                      reverse=True)
+
+    def create(self, cr, user, vals, context=None):
+        if context is None:
+            context = {}
+        if context and context.get('bve'):
+            vals['state'] = 'base'
+        res = super(IrModel, self).create(cr, user, vals, context)
+        if vals.get('state', 'base') == 'bve':
+            vals['state'] = 'manual'
+
+            # add model in registry
+            self.instanciate(cr, user, vals['model'], context)
+            self.pool.setup_models(cr, partial=(not self.pool.ready))
+
+            # update database schema
+            # model = self.pool[vals['model']]
+            # ctx = dict(
+            #     context,
+            #     field_name=vals['name'],
+            #     field_state='manual',
+            #     select=vals.get('select_level', '0'),
+            #     update_custom_fields=True)
+            RegistryManager.signal_registry_change(cr.dbname)
+
+#        self.write(cr, user, [res], {'state': 'manual'})
+        q = ("""UPDATE ir_model SET state = 'manual'
```

Comment:
Thanks for the reply: my technical level is not good enough here to challenge!
Is there any workaround: @pedrobaeza @dreispt ?